src/components/views/hotelOwnerView/steps/RoomsStep.jsx

```diff
@@ -0,0 +1,189 @@
+import React, { useState, useEffect } from 'react'
+import { makeStyles } from '@material-ui/core/styles'
+import Snackbar from '@material-ui/core/Snackbar'
+import MuiAlert from '@material-ui/lab/Alert'
+import TextField from '@material-ui/core/TextField'
+import Button from '@material-ui/core/Button'
+import List from '@material-ui/core/List'
+import ListItem from '@material-ui/core/ListItem'
+import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction'
+import ListItemText from '@material-ui/core/ListItemText'
+import IconButton from '@material-ui/core/IconButton'
+import DeleteIcon from '@material-ui/icons/Delete'
+
+const useStyles = makeStyles((theme) => ({
+  form: {
+    '& > *': {
+      margin: theme.spacing(1),
+      width: '25ch',
+    },
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  list: {
+    width: '12rem',
+    padding: '1rem',
+    backgroundColor: theme.palette.background.paper,
+  },
+}))
+
+function Alert(props) {
+  return <MuiAlert elevation={6} variant="filled" {...props} />
+}
+
+const useForceUpdate = () => {
+  const [value, setValue] = useState(0)
+  return () => setValue((value) => value + 1)
+}
+
+export const RoomsStep = ({ setRooms }) => {
```

Comment:
actually you can make it as a class, this is very extensive function so I would suggest to do it as a class (your decision) just let me know what you decide